apps/etl-func/src/domain/interfaces/tokenizer.ts

```diff
@@ -1,8 +1,9 @@
+import { FiscalCode } from "../message-status.js";
+
 export interface TokenizerClient {
   tokenize(pii: string): Promise<string>;
 }
-
-export const maskSensitiveInfo =
-  (fiscalCode: string) =>
-  async (client: TokenizerClient): Promise<string> =>
-    await client.tokenize(fiscalCode);
+export interface RecipientRepository {
+  get(fiscalCode: FiscalCode): Promise<string | undefined>;
+  upsert(fiscalCode: FiscalCode, recipientId: string): Promise<void>;
+}
```

Comment:
I don't understand why you created a `RecipientRepository` - wasn't `maskSensitiveInfo` the right abstraction?

What is the purpose of this repository?

Reply:
this repository is the one responsible for the cached recipientId, I use it to check if a fiscalCode is already cached or not and to save the tokenized fiscalCode in the cache [here](https://github.com/pagopa/io-messages/pull/66/files#diff-e05293644adfa971f457c46d7f8998a3769bcd4dde3aae32e8359b548b2b338f:~:text=const%20recipientId%20%3D%20await%20this.%23recipientRepo.get(fiscalCode)%3B). This repository has the specific implementation for redis. 

The maskSensitiveInfo method I think was an over-engineering since it was just one line of code and didn't do anything that the client didn't do already.

Reply:
the fact that a certain fiscalCode is cached or not is just an implementation detail, and should not be important (and known) at the domain level

the domain just `tokenize` the fiscalCode, how is tokenized (by PDV, cached or not) is relevant only to infrastructure layer